### DISCUSSION TITLE
Subscribe to hashtags across all channels

### DISCUSSION THREAD

**Discussion by emilyacook** [mattermost/community#40] [Moonshot Ideas]
So the problem I have, that I think this could solve, is twofold:

One: if you are not a member of a channel, or unless someone @ mentions you in a comment and chooses to invite you to that channel, it's basically like that channel doesn't exist to you. You're completely blind regarding any discussion going on in there.

Two, which is kinda the flip-side to the first one: for channels that you're interested in, but not very active in, you basically have to scan every message to find those few things that might be relevant to you. I dont know about yall, but for me that results in me joining a bunch of channels, maybe actively checking them for a day or two, until eventually (a) they just kinda sit there until I decide to do "jump to latest" from time to time and look at a few messages until the cycle starts again, or (b) am forced to do a purge because I'm just in too many channels I don't check. And then # 1 happens, where the discussions in that channel now basically don't exist for me.

I believe hashtags were probably created to combat # 2, but in my opinion they are woefully underutilized. Right now they pretty much only get used consistently for queueing items for meetings (like in the Dev meeting).

What could potentially solve both problems is the ability to subscribe to certain hashtags, and then get notified of them even in channels you aren't a member of.

I envisioning this looking like a mashup of collapsed reply threads and channel categories, where there's a folder at the top for all of the hashtags you're subscribed to (like categories), and you click each one and it's a list of the messages that have used those hashtags (like threads)
<img width="236" alt="Screen Shot 2021-06-22 at 11 39 19 AM" src="https://user-images.githubusercontent.com/35075700/122966754-4e08aa80-d34f-11eb-9df1-55c657637052.png">

I may have shamelessly stolen the design idea from [Zulip](https://zulip.com/)

**Comment by esethna** [comment]
Interesting idea! I think one pre-requisite to this is hashtag autocomplete so users know what tags are being used and can reuse popular ones: https://mattermost.atlassian.net/browse/MM-10456

**Comment by Willyfrog** [comment]
this would be great if it was also managed from the global threads, so I get all things that might interest me in the same place

**Comment by chetanyakan** [comment]
Also AI-generated hashtags: https://github.com/mattermost/community/discussions/100

**Comment by darkLord19** [comment]
I think it'd be technically challenging but what if we could subscribe to any keyword and not just hashtags!

**Reply by esethna** [reply]
You can actually already do that in a way with "keywords that trigger mentions" in account settings

**Reply by darkLord19** [reply]
Does that work even when you're not part of the channel where the keyword was mentioned?

**Reply by Willyfrog** [reply]
No, the closest is an at-mention that suggest to invite the user. But if people decide to not add you, you don't get the mention

**Reply by esethna** [reply]
Nope, good point @darkLord19. Since you personal words that trigger mentions are not shared publicly
